Framework/Algorithms/src/Q1DWeighted.cpp

```diff
@@ -162,6 +207,167 @@ void Q1DWeighted::bootstrap(const MatrixWorkspace_const_sptr &inputWS) {
+      viewportParams["Rotation"][2] != 1.) {
+    g_log.warning("The shapes were created using a rotated viewport not using "
+                  "Z- projection, which is not supported. Results are likely "
+                  "to be erroneous.");
+  }
+}
+
+/**
+ * @brief Q1DWeighted::getSectorParams
+ * @param params the vector of strings containing the data defining the sector
+ * @param viewport the previously created map of the viewport's parameters
+ */
+void Q1DWeighted::getSectorParams(
+    std::vector<std::string> &params,
+    std::map<std::string, std::vector<double>> &viewport) {
+  double zoom = viewport["Zoom"][0];
+
+  double innerRadius = std::stod(params[1]) / zoom;
+  double outerRadius = std::stod(params[2]) / zoom;
+
+  double startAngle = std::stod(params[3]);
+  double endAngle = std::stod(params[4]);
+
+  double centerAngle = (startAngle + endAngle) / 2;
+  if (endAngle < startAngle)
+    centerAngle = fmod(centerAngle + M_PI, 2 * M_PI);
+
+  double angleRange = std::fmod(endAngle - startAngle, 2 * M_PI);
+  angleRange = angleRange >= 0 ? angleRange : angleRange + 2 * M_PI;
+
+  // since the viewport was in Z-, the axis are inverted so we have to take the
+  // symmetry of the angle
+  centerAngle = fmod(3 * M_PI - centerAngle, 2 * M_PI);
+
+  double xOffset = viewport["Translation"][0];
+  double yOffset = viewport["Translation"][1];
+
+  double centerX = -(std::stod(params[5]) - xOffset) / zoom;
+  double centerY = (std::stod(params[6]) - yOffset) / zoom;
+
+  if (m_asymmWedges ||
+      !checkIfSymetricalWedge(innerRadius, outerRadius, centerX, centerY,
+                              centerAngle, angleRange)) {
+
+    m_wedgesInnerRadius.push_back(innerRadius);
+    m_wedgesOuterRadius.push_back(outerRadius);
+    m_wedgesCenterAngle.push_back(centerAngle);
+    m_wedgesAngleRange.push_back(angleRange);
+    m_wedgesCenterX.push_back(centerX);
+    m_wedgesCenterY.push_back(centerY);
+    m_wedgesIsAsymmetric.push_back(true);
+  }
+}
+
+/**
+ * @brief Q1DWeighted::checkIfSymetricalWedge
+ * Check if the symetrical wedge to the one defined by the parameters is already
+ * registered in the parameter list
+ * @param innerRadius the inner radius
+ * @param outerRadius the outer radius
+ * @param centerX the x position of the center
+ * @param centerY the y position of the center
+ * @param centerAngle the center of the wedge
+ * @param angleRange the angular range of the wedge
+ * @return true if a symetrical wedge already exists
+ */
+bool Q1DWeighted::checkIfSymetricalWedge(double innerRadius, double outerRadius,
+                                         double centerX, double centerY,
+                                         double centerAngle,
+                                         double angleRange) {
+
+  for (size_t i = 0; i < m_wedgesInnerRadius.size(); ++i) {
+    double diffAngle = fabs(fmod(centerAngle - m_wedgesCenterAngle[i], M_PI));
+
+    if (innerRadius == m_wedgesInnerRadius[i] &&
```

Comment:
This condition is not readable, please simplify.

Reply:
Also the purpose is not clear, maybe it would be if it is simpler.
Some properties you compare with tolerance, some without. 
In general strict equality checks on doubles is not recommended.
Define a single constant tolerance at the top, and compare all with that.